nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/ProcessorController.java

```diff
@@ -1126,6 +1137,29 @@ public void changed(ObservableValue<? extends Number> observableValue, Number nu
         });
         initTable();
 
+        rowToggleGroup.selectedToggleProperty().addListener(e -> handleRowDimChange());
+        vecNumListener = (observableValue, string, string2) -> {
+            String text = realImagChoiceBox.getValue();
+            int vecNum = realImagChoices.indexOf(text);
+            chartProcessor.setVector(vecNum);
+            setFileIndex();
+        };
+
+    }
+
+    /**
+     * Listener for the Processor availability status and updates the processor available status
+     * @param newStatus the newly updated status
+     */
+    public void processorAvailableStatusUpdated(Boolean newStatus) {
+        processorAvailable.setValue(newStatus);
+    }
```

Comment:
Used observer pattern to disable "Process" button when mutliple processor controllers were created